### PR TITLE
Ins 2856 seedpulse and reason

### PR DESCRIPTION
**Pull request by ilyapt**
**- What I did**
- SeedManager stores pulse for seed
- ContractRequester uses seed pulse and hash of api request to makes Reason

### CODE REVIEW

File: api/call.go [hunk 1]
```diff
@@ -201,15 +202,17 @@ func (ar *Runner) callHandler() func(http.ResponseWriter, *http.Request) {
 			return
 		}
 
-		if err := ar.checkSeed(contractRequest.Params.Seed); err != nil {
+		seedPulse, err := ar.checkSeed(contractRequest.Params.Seed)
+		_ = seedPulse
```
Comment (ivanshibitov) [inline api/call.go]: mmmm?

File: contractrequester/contractrequester.go [hunk 2]
```diff
@@ -82,6 +82,15 @@ func randomUint64() uint64 {
...
 func (cr *ContractRequester) SendRequest(ctx context.Context, ref *insolar.Reference, method string, argsIn []interface{}) (insolar.Reply, error) {
+	pulse, err := cr.PulseAccessor.Latest(ctx)
+	if err != nil {
+		return nil, errors.Wrap(err, "[ ContractRequester::SendRequest ] Couldn't fetch current pulse")
+	}
+	return cr.SendRequestWithPulse(ctx, ref, method, argsIn, pulse.PulseNumber)
+}
+
+func (cr *ContractRequester) SendRequestWithPulse(ctx context.Context, ref *insolar.Reference, method string, argsIn []interface{}, pulse insolar.PulseNumber) (insolar.Reply, error) {
+	// func (cr *ContractRequester) SendRequest(ctx context.Context, ref *insolar.Reference, method string, argsIn []interface{}) (insolar.Reply, error) {
```
Comment (ivanshibitov) [inline contractrequester/contractrequester.go]: mmmmm?

File: api/seedmanager/seedmanager.go [hunk 3]
```diff
@@ -30,11 +32,16 @@ const DefaultTTL = 5 * time.Second
 // DefaultCleanPeriod default time period for launching cleaning goroutine
 const DefaultCleanPeriod = 5 * time.Second
 
+type storedSeed struct {
+	expire Expiration
```
Comment (ivanshibitov) [inline api/seedmanager/seedmanager.go]: expire -> expiration

Comment (ilyapt) [reply api/seedmanager/seedmanager.go]: done

File: contractrequester/contractrequester.go (hunk 2 above)
Comment (ilyapt) [reply contractrequester/contractrequester.go]: removed

File: api/call.go (hunk 1 above)
Comment (ilyapt) [reply api/call.go]: removed

**Pull request merged**